### PR TITLE
Loggregator improvements and tests

### PR DESCRIPTION
**Pull request by stefanschneider**
Tested on bosh-lite + https://github.com/cloudfoundry/cf-release/tree/50381ee2b4a12e24e1ef1663bde1f57847d67337 
 
v3 push was failing to breaking changes.

### CODE REVIEW

**Comment by stefanschneider** [comment]
@florindragos  @mihaibuzgau  can you please review this?

**Comment by florindragos** [comment]
verify this please

**Comment by florindragos** [comment]
verify this please

**Comment by stefanschneider** [comment]
verify this please

**Comment by stefanschneider** [comment]
verify this please

File: src/CloudFoundry.CloudController.Test.Integration/LogyardTest.cs [hunk 1]
```diff
@@ -115,6 +115,11 @@ public void LogsTest()
                 }
             }
 
+            if (client.Info.GetV1Info().Result.AppLogEndpoint == null)
+            {
+                Assert.Inconclusive("CloudFoundry target does not have a logyard endpoint");
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.Test.Integration/LogyardTest.cs]: We've had issues with inconclusive - I remember using categories to decide which tests we want to run against a certain target?

File: src/CloudFoundry.CloudController.Test.Integration/LoggregatorTest.cs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            if (client.Info.GetInfo().Result.LoggingEndpoint == null)
+            {
+                Assert.Inconclusive("CloudFoundry target does not have a loggregator endpoint");
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.Test.Integration/LoggregatorTest.cs]: We've had issues with inconclusive - I remember using categories to decide which tests we want to run against a certain target?

File: src/CloudFoundry.CloudController.Test.Integration/LoggregatorTest.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        [TestCategory("RequiresLoggregator")]
+        public void LoggregatorTailTest()
+        {
+            CreateAppResponse app = client.Apps.CreateApp(apprequest).Result;
+
+            Guid appGuid = app.EntityMetadata.Guid;
+
+            if (client.Info.GetInfo().Result.LoggingEndpoint == null)
+            {
+                Assert.Inconclusive("CloudFoundry target does not have a loggregator endpoint");
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.Test.Integration/LoggregatorTest.cs]: We've had issues with inconclusive - I remember using categories to decide which tests we want to run against a certain target?

File: src/CloudFoundry.CloudController.Test.Integration/PushTest.cs [hunk 4]
```diff
@@ -63,7 +63,7 @@ public static void ClassInit(TestContext context)
 
             if (winStack == Guid.Empty)
             {
-                throw new Exception("Could not test on a deployment without a windows 2012 stack");
+                Assert.Inconclusive("Could not test on a deployment without a windows 2012 stack");
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.Test.Integration/PushTest.cs]: We've had issues with inconclusive - I remember using categories to decide which tests we want to run against a certain target?

Comment (stefanschneider) [inline src/CloudFoundry.CloudController.Test.Integration/PushTest.cs]: This is just when running the test on a local VS + bosh-lite. The CI has all the categories needed.

**Comment by viovanov** [comment]
:+1:

**Pull request merged**